src/test/java/fr/inria/diversify/automaticbuilder/GradleAutomaticBuilderTest.java

```diff
@@ -29,39 +36,85 @@
     }
 
     @Test
-    public void buildClasspath() throws Exception {
+    public void buildClasspath_whenAppliedToDSpotTestExampleProject() throws Exception {
+        Log.info("Starting Gradle Automatic Builder buildClasspath() test...");
+
+        String[] expectedCompilationLibraries = {"log4j-api-2.8.2.jar", "log4j-core-2.8.2.jar"};
+        String[] expectedTestCompilationLibraries = {"hamcrest-core-1.3.jar", "junit-4.12.jar"};
+
         String classPath = sut.buildClasspath("src/test/resources/test-projects/");
 
         assertNotNull(classPath, "Classpath should be null");
 
-        assertTrue("Classpath should contain gradle-pitest-plugin-1.1.11.jar library as copmile/runtime dependency", classPath.contains("gradle-pitest-plugin-1.1.11.jar")); // compile dependency
+        assertTrue("Classpath should contain " + expectedCompilationLibraries[0] + " library as compile/runtime dependency", classPath.contains(expectedCompilationLibraries[0])); // compile dependency
+        assertTrue("Classpath should contain " + expectedCompilationLibraries[1] + " library as compile/runtime dependency", classPath.contains(expectedCompilationLibraries[1])); // compile dependency
+
+        assertTrue("Classpath should contain " + expectedTestCompilationLibraries[0] + " library as test dependency", classPath.contains(expectedTestCompilationLibraries[0])); // test compile dependency
+        assertTrue("Classpath should contain " + expectedTestCompilationLibraries[1] + " library as test dependency", classPath.contains(expectedTestCompilationLibraries[1])); // test compile dependency
 
-        assertTrue("Classpath should contain hamcrest-core-1.3.jar library as test dependency", classPath.contains("hamcrest-core-1.3.jar")); // test compile dependency
-        assertTrue("Classpath should contain junit-4.12.jar library as test dependency", classPath.contains("junit-4.12.jar")); // test compile dependency
+        Log.info("Gradle Automatic Builder buildClasspath() test complete.");
     }
 
     @Test
-    public void runPit() throws Exception {
+    public void runPit_whenNoTestClassIsSpecified() throws Exception {
+        Log.info("Starting Gradle Automatic Builder runPit() test when no test class is specified...");
+
+        List<PitResult> pitResults = sut.runPit("src/test/resources/test-projects/");
+
+        assertTrue("PIT results shouldn't be null", pitResults != null);
+        assertTrue("PIT results shouldn't be empty", !pitResults.isEmpty());
+
+        Log.info("Gradle Automatic Builder runPit() test complete when no test class is specified.");
     }
 
     @Test
-    public void runPit1() throws Exception {
+    public void runPit_whenTestClassIsSpecified() throws Exception {
+        Log.info("Starting Gradle Automatic Builder runPit() test when a test class is specified...");
+
+        CtClass<Object> testClass = Utils.getInputProgram().getFactory().Class().get("example.TestSuiteExample");
```

Comment:
I'm sure that this CtClass is null if you do not use Utils.getInputConfiguration in the set up, and it results that this test is the same than above.